tests/zfs-tests/tests/functional/cli_root/zfs_mount/zfs_mount_all_002_pos.ksh

```diff
@@ -0,0 +1,209 @@
+set -A vol "$TESTVOL" "$TESTVOL1"
+
+# Test the mounted state of root dataset (testpool/testctr)
+typeset mnt=$TESTCTR
+
+function setup_all
+{
+	typeset -i i=0
+	typeset -i j=0
+	typeset path
+
+	while (( i < ${#ctr[*]} )); do
+
+		path=${TEST_BASE_DIR%%/}/testroot$$/$TESTPOOL
+		if [[ -n ${ctr[i]} ]]; then
+			path=$path/${ctr[i]}
+
+			setup_filesystem "$DISKS" "$TESTPOOL" \
+				"${ctr[i]}" "$path" \
+				"ctr"
+		fi
+
+		if is_global_zone ; then
+			j=0
+			while (( j < ${#vol[*]} )); do
+				setup_filesystem "$DISKS" "$TESTPOOL" \
+					"${ctr[i]}/${vol[j]}" \
+					"$path/${vol[j]}" \
+					"vol"
+				((j = j + 1))
+			done
+		fi
+
+		j=0
+		while (( j < ${#fs[*]} )); do
+			setup_filesystem "$DISKS" "$TESTPOOL" \
+				"${ctr[i]}/${fs[j]}" \
+				"$path/${fs[j]}"
+			((j = j + 1))
+		done
+
+		((i = i + 1))
+	done
+
+	return 0
+}
+
+function cleanup_all
+{
+	typeset -i i=0
+	typeset -i j=0
+	typeset path
+
+	((i = ${#ctr[*]} - 1))
+
+	while (( i >= 0 )); do
+		if is_global_zone ; then
+			j=0
+			while (( j < ${#vol[*]} )); do
+				cleanup_filesystem "$TESTPOOL" \
+					"${ctr[i]}/${vol[j]}"
+				((j = j + 1))
+			done
+		fi
+
+		j=0
+		while (( j < ${#fs[*]} )); do
+			cleanup_filesystem "$TESTPOOL" \
+				"${ctr[i]}/${fs[j]}"
+			((j = j + 1))
+		done
+
+		[[ -n ${ctr[i]} ]] && \
+			cleanup_filesystem "$TESTPOOL" "${ctr[i]}"
+
+		((i = i - 1))
+	done
+
+	[[ -d ${TEST_BASE_DIR%%/}/testroot$$ ]] && \
+		rm -rf ${TEST_BASE_DIR%%/}/testroot$$
+}
+
+#
+# This function takes a single true/false argument:
+# - true:	it verifies that the $mnt file system is mounted.
+# - false:	it verifies that the $mnt file system is unmounted.
+#
+# In both scenarios, it ensures that the file systems in fs1 remain unmounted.
+#
+function verify_related
+{
+	typeset -i i=0
+	typeset -i j=0
+	typeset path
+	typeset logfunc
+
+	while (( i < ${#ctr[*]} )); do
+
+		if $1 && { [[ ${ctr[i]} == $mnt ]] || [[ ${ctr[i]} == $mnt/* ]] }; then
+			logfunc=log_must
+		else
+			logfunc=log_mustnot
+		fi
+
+		path=$TESTPOOL
+		[[ -n ${ctr[i]} ]] && \
+			path=$path/${ctr[i]}
+
+		if is_global_zone ; then
+			j=0
+			while (( j < ${#vol[*]} )); do
+				log_mustnot mounted "$path/${vol[j]}"
+				((j = j + 1))
+			done
+		fi
+
+		j=0
+		while (( j < ${#fs[*]} )); do
+			$logfunc mounted "$path/${fs[j]}"
+			((j = j + 1))
+		done
+
+		$logfunc mounted "$path"
+
+		((i = i + 1))
+	done
+
+	return 0
+}
+
+
+log_assert "Verify that 'zfs $mountall filesystem' succeeds as root, " \
+	"and all available related ZFS filesystems are mounted."
+
+log_onexit cleanup_all
+
+log_must setup_all
+
+export __ZFS_POOL_RESTRICT="$TESTPOOL"
+log_must zfs $unmountall
+unset __ZFS_POOL_RESTRICT
+
+verify_related false
+
+export __ZFS_POOL_RESTRICT="$TESTPOOL"
+log_must zfs $mountall $TESTPOOL/$mnt
+unset __ZFS_POOL_RESTRICT
+
+verify_related true
+
+log_note "Verify that 'zfs $mountcmd' will display " \
+	"all ZFS filesystems related to '$TESTPOOL/$mnt' are currently mounted."
```

Comment:
`all ZFS filesystems related to` should be `all available ZFS filesystems related to`